internal/catalog/internal/types/health_status_test.go

```diff
@@ -214,3 +215,88 @@ func TestValidateHealthStatus_InvalidOwner(t *testing.T) {
 		})
 	}
 }
+
+func TestHealthStatusACLs(t *testing.T) {
+	registry := resource.NewRegistry()
+	Register(registry)
+
+	workload := resourcetest.Resource(pbcatalog.WorkloadType, "test").ID()
+	node := resourcetest.Resource(pbcatalog.NodeType, "test").ID()
+
+	healthStatusData := &pbcatalog.HealthStatus{
+		Type:   "tcp",
+		Status: pbcatalog.Health_HEALTH_PASSING,
+	}
+
+	cases := map[string]resourcetest.ACLTestCase{
+		"no rules": {
+			Rules:   ``,
+			Data:    healthStatusData,
+			Owner:   workload,
+			Typ:     pbcatalog.HealthStatusType,
+			ReadOK:  resourcetest.DENY,
+			WriteOK: resourcetest.DENY,
+			ListOK:  resourcetest.DEFAULT,
+		},
+		"service test read": {
+			Rules:   `service "test" { policy = "read" }`,
+			Data:    healthStatusData,
+			Owner:   workload,
+			Typ:     pbcatalog.HealthStatusType,
+			ReadOK:  resourcetest.ALLOW,
+			WriteOK: resourcetest.DENY,
+			ListOK:  resourcetest.DEFAULT,
+		},
+		"service test write": {
+			Rules:   `service "test" { policy = "write" }`,
+			Data:    healthStatusData,
+			Owner:   workload,
+			Typ:     pbcatalog.HealthStatusType,
+			ReadOK:  resourcetest.ALLOW,
+			WriteOK: resourcetest.ALLOW,
+			ListOK:  resourcetest.DEFAULT,
+		},
+		"service test read with node owner": {
+			Rules:   `service "test" { policy = "read" }`,
+			Data:    healthStatusData,
+			Owner:   node,
+			Typ:     pbcatalog.HealthStatusType,
+			ReadOK:  resourcetest.DENY,
+			WriteOK: resourcetest.DENY,
+			ListOK:  resourcetest.DEFAULT,
+		},
+		"service test write with node owner": {
+			Rules:   `service "test" { policy = "write" }`,
+			Data:    healthStatusData,
+			Owner:   node,
+			Typ:     pbcatalog.HealthStatusType,
+			ReadOK:  resourcetest.DENY,
+			WriteOK: resourcetest.DENY,
+			ListOK:  resourcetest.DEFAULT,
+		},
+		"node test read with node owner": {
+			Rules:   `node "test" { policy = "read" }`,
+			Data:    healthStatusData,
+			Owner:   node,
+			Typ:     pbcatalog.HealthStatusType,
+			ReadOK:  resourcetest.ALLOW,
+			WriteOK: resourcetest.DENY,
+			ListOK:  resourcetest.DEFAULT,
+		},
+		"node test write with node owner": {
```

Comment:
It might be nice to add test cases that node read and write don't grant permissions when the owner is a workload.